packages/frontend/src/components/common/PivotTable/index.tsx

```diff
@@ -158,6 +181,25 @@ const PivotTable: FC<PivotTableProps> = ({
 
     const hasRowTotals = data.pivotConfig.rowTotals;
 
+    const rowVirtualizer = useVirtual({
+        parentRef: containerRef,
+        size: data.dataValues.length,
+        overscan: 10,
+    });
+    const { virtualItems: virtualRows, totalSize } = rowVirtualizer;
+    const paddingTop =
+        virtualRows.length > 0 ? virtualRows?.[0]?.start || 0 : 0;
+    const paddingBottom =
+        virtualRows.length > 0
+            ? totalSize - (virtualRows?.[virtualRows.length - 1]?.end || 0)
+            : 0;
+    const cellsCount =
+        (hideRowNumbers ? 0 : 1) +
+        (data.indexValueTypes.length === 0 ? data.titleFields[0].length : 0) +
+        data.indexValueTypes.length +
+        data.dataValues[0].length +
+        (hasRowTotals && data.rowTotals ? data.rowTotals[0].length : 0);
+
```

Comment:
(you can ignore this) it might be better if we expose this info from `pivotQueryResults`. `tableShape: [x, y]`